pandas/io/json/_json.py

```diff
@@ -750,6 +740,8 @@ def read_json(
 }}\
 '
     """
+    if isinstance(path_or_buf, str):
```

Comment:
I think this check is probably too restrictive. I think we want to allow
```
pd.read_json("path/to/file.json")
```

Going forward. We just don't want
```
pd.read_json("{json: object}") # or however it's spelt
```

So rather than checking here. The deprecation probably wants enforced in `JsonReader`, perhaps like this?
```patch
diff --git a/pandas/io/json/_json.py b/pandas/io/json/_json.py
index 8775c65f14..78f541d3df 100644
--- a/pandas/io/json/_json.py
+++ b/pandas/io/json/_json.py
@@ -18,6 +18,7 @@ from typing import (
     TypeVar,
     overload,
 )
+import warnings
 
 import numpy as np
 
@@ -30,6 +31,7 @@ from pandas._libs.tslibs import iNaT
 from pandas.compat._optional import import_optional_dependency
 from pandas.errors import AbstractMethodError
 from pandas.util._decorators import doc
+from pandas.util._exceptions import find_stack_level
 from pandas.util._validators import check_dtype_backend
 
 from pandas.core.dtypes.common import ensure_str
@@ -916,7 +918,14 @@ class JsonReader(abc.Iterator, Generic[FrameSeriesStrT]):
             and not file_exists(filepath_or_buffer)
         ):
             raise FileNotFoundError(f"File {filepath_or_buffer} does not exist")
-
+        else:
+            warnings.warn(
+                "Passing literal json to 'read_json' is deprecated and "
+                "will be removed in a future version. To read from a "
+                "literal string, wrap it in a 'StringIO' object.",
+                FutureWarning,
+                stacklevel=find_stack_level(),
+            )
         return filepath_or_buffer
 
     def _combine_lines(self, lines) -> str:
```
WDYT?

Reply:
I see the value in doing that. Though, that doesn't seem to be what our repo maintainers are going after:

Reply:
Ahh okay! I was thinking that they would want to turn strings into StringIO objects! And yeah I definitely understand that we would want to have at least one release with a deprecation warning! 💯